docs/OnChainSettlement.md

```diff
@@ -0,0 +1,88 @@
+# Report notarization and on-chain settlement of outbound transactions
+
+On this page we describe how the Spectrum-protocol settles outbound
+transactions on-chain. We define an _outbound transaction_ as one that
+transfers value from Spectrum to a user address in $S_k$.
+
+Let us assume that the Spectrum protocol is running in the Normal Flow and
+consider a particular external system $S_k$ during epoch $e_n$ with a committee
+denoted by $V_n^k$ . For an arbitrary slot in this epoch, suppose that the
+committee members have been observing events in $S_k$  and the L+ mempool,
+leading to the slot leader to propose a batch or report of events in $S_k$ that
+we denote by $r^*$. The report $r^*$ consists of:
+ - The _progress point_ of system $S_k$.
+ - A list of effects on $S_k$. There are 4 types of effects:
+     1. Importation of value from $S_k$ into Spectrum's on-chain vault in $S_k$
+     (to be defined below).
+     2. Exportation of value from Spectrum's on-chain vault to a user address
+     in $S_k$.
+     3. Revocation of a previously imported value due to a chain rollback in
+     $S_k$.
+     4. Notification that $S_k$ has reached a particular progress point (e.g.
+     new block height).
+ - The hash value of the root of a Merkle tree formed by taking the above list
+   of effects as leaf nodes. Note that we must define a partial order on these
+   effects to enable proper verification (see appendix A).
+
+### Report notarization
+
+Once the slot leader has proposed a report $r^*$, it is disseminated to all
+committee members. Each committee member will recreate the Merkle tree from
+these effects and confirms that the root hash equals the hash within $r^*$.
+Then using $\mathcal{F}_{SIG}$, the committee computes the aggregated signature
+of the root hash value. We can then define the _notarized report_ $R^*$ as
+$r^*$ appended with the aggregated signature. $R^*$ is disseminated to all
+committee members.
+
+```rust
+struct NotarizedReport {
+    certificate: ReportCertificate,
+    value_to_export: Vec<(TermCell, rs_merkle::MerkleProof)>,
+}
+```
+
+### Settlement of export-of-value effects on $S_k$
+
+Let $I^k_0, I^k_1, \ldots, I^k_m$  denote the _ordered_ cell ids of all
+export-of-value effects in $R^*$ that are intended to be executed on $S_k$.
+Such effects are represented by a cell id $I^k_j$ that is associated with a
+_terminal_ cell $C^k_j$ in the system. Upon receiving the notarized report
+$R^*$, the $S_k$-vault forms the associated transactions on-chain which are
```

Comment:
Who pays TX fees on chain?

Reply:
Also, not clear how `$S_k$-vault` can form on-chan TX itself. It's just a passive pool of assets guarded with a specific script.

Reply:
> Who pays TX fees on chain?

The spectrum user who initiated the transaction. I think we can add a `tx_fee` field to `TermCell`, so that it must be decided when creating a Spectrum transaction. We could give the user the option of paying fees in different coins, like how we currently do on the DEX (in erg or in SPF). Or perhaps all fees are specified in SPF, and the network computes 
and pays the TX fee per-chain when processing outbound TX.

> Also, not clear how `$S_k$-vault` can form on-chan TX itself. It's just a passive pool of assets guarded with a specific script.

There definitely needs to be an off-chain component that creates the TXs. It can still be part of the implementation of the `Vault` trait. What do you think?

Reply:
Yes, sounds good.